core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -578,9 +578,18 @@ private[spark] class BlockManager(
     doGetRemote(blockId, asBlockResult = false).asInstanceOf[Option[ByteBuffer]]
   }
 
+  private def getLocations(blockId: BlockId): Seq[BlockManagerId] = {
+    // Since block managers can share an identical host, we put these preferred
+    // locations first.
```

Comment:
I would move this to the java doc and rephrase it
```
/**
 * Return a list of locations for the given block, prioritizing the local machine since multiple
 * block managers can share the same host.
 */
```